app/web/src/features/communities/CommunityBase.tsx

```diff
@@ -96,58 +54,5 @@ export default function CommunityBase({
       </Alert>
     );
 
-  return (
-    <div className={classes.root}>
-      <HeaderImage community={community} className={classes.header} />
-      <div className={classes.breadcrumbsContainer}>
-        <Breadcrumbs aria-label="breadcrumb" className={classes.breadcrumbs}>
-          {community.parentsList
-            .map((parent) => parent.community)
-            .filter(
-              (communityParent): communityParent is CommunityParent.AsObject =>
-                !!communityParent
-            )
-            .map((communityParent, index, array) =>
-              index === array.length - 1 ? (
-                <Typography
-                  variant="body1"
-                  color="textPrimary"
-                  key={`breadcrumb-${communityParent?.communityId}`}
-                >
-                  {communityParent.name}
-                </Typography>
-              ) : (
-                <MuiLink
-                  component={Link}
-                  to={routeToCommunity(
-                    communityParent.communityId,
-                    communityParent.slug
-                  )}
-                  key={`breadcrumb-${communityParent?.communityId}`}
-                >
-                  {communityParent.name}
-                </MuiLink>
-              )
-            )}
-        </Breadcrumbs>
-        <JoinCommunityButton community={community} />
-      </div>
-      <TabContext value={tab}>
-        <TabBar
-          ariaLabel={COMMUNITY_TABS_A11Y_LABEL}
-          setValue={(newTab) =>
-            history.push(
-              `${routeToCommunity(
-                community.communityId,
-                community.slug,
-                newTab === "overview" ? undefined : newTab
-              )}`
-            )
-          }
-          labels={communityTabBarLabels}
-        />
-      </TabContext>
-      {children({ community, communitySlug })}
-    </div>
-  );
+  return <div className={classes.root}>{children({ community })}</div>;
```

Comment:
The reason this was using a render prop was because the header was assumed to be the same across the pages that use this component. But since you've now stripped everything out to allow more flexibility, the only thing left that's worth sharing is the stateful logic in here, and so there's much point in keeping `CommunityBase`.

We may as well everything out here and turn it all into a custom hook that can be used everywhere else?

Reply:
Ahh actually scrap what I said, there are still some error handling logic here so maybe this component isn't completely useless 😅